models/repo_generate.go

```diff
@@ -69,7 +98,12 @@ func generateRepoCommit(e Engine, repo, templateRepo, generateRepo *Repository,
 		return fmt.Errorf("remove git dir: %v", err)
 	}
 
-	// Variable expansion in _template files
+	// Variable expansion
+	globs, err := parseGiteaTemplate(tmpDir)
+	if err != nil {
+		return fmt.Errorf("parseGiteaTemplate: %v", err)
```

Comment:
I think this should be a nice, translatable error at the top of the page (e.g. _"Your repository couldn't be created because the template has errors in its `/.giteatemplate` directive"_). It can be quite frustrating to debug otherwise.

Why this error and not a `git` or database failure for example? A `git` failure is a system problem, but an error that prevents `.giteatemplate` from parsing is a user error.

Reply:
Currently the only errors coming out of this function is file-errors. Glob errors are skipped and logged in console.  
Should this collect errors to present to the user? Further, should glob errors completely stop generation?

Reply:
IMHO, errors due to incorrect usage (e.g. badly parametrized template by the creator) should never generate a 500 Server Error but a friendly message for the user. Other types of error (file errors, db errors, etc.) should generate a 500 Server Error.

Regarding this line of code in particular, I thought that gobwas might return an error due to bad syntax, but I can't think of an example that would cause that. So you can ignore me. 😄